functions/go/fix/fixpkg/fix.go

```diff
@@ -0,0 +1,453 @@
+		}
+
+		if defExt.Substitution != nil {
+			// parse recursively if it reference is substitution
+			substVal, err := s.substituteUtil(defExt, visited)
+			if err != nil {
+				return "", err
+			}
+			pattern = strings.ReplaceAll(pattern, v.Marker, substVal)
+			continue
+		}
+
+		if val, found := defExt.Setter.EnumValues[defExt.Setter.Value]; found {
+			// the setter has an enum-map. we should replace the marker with the
+			// enum value looked up from the map rather than the enum key
+			pattern = strings.ReplaceAll(pattern, v.Marker, val)
+		} else {
+			pattern = strings.ReplaceAll(pattern, v.Marker, fmt.Sprintf("${%s}", defExt.Setter.Name))
+		}
+	}
+	return pattern, nil
+}
+
+// listSetters extracts the setters information from the input Kptfile node
+func listSetters(object *yaml.RNode) (map[string]string, error) {
+	setters := make(map[string]string)
+	// read the OpenAPI definitions in Kptfile
+	def, err := object.Pipe(yaml.LookupCreate(yaml.MappingNode, "openAPI", "definitions"))
+	if err != nil {
+		return nil, err
+	}
+	if yaml.IsMissingOrNull(def) {
+		return nil, nil
+	}
+
+	// iterate over definitions -- find those that are setters
+	err = def.VisitFields(func(node *yaml.MapNode) error {
+		setter := setters2.SetterDefinition{}
+
+		// the definition key -- contains the setter name
+		key := node.Key.YNode().Value
+
+		if !strings.HasPrefix(key, fieldmeta.SetterDefinitionPrefix) {
+			// not a setter -- doesn't have the right prefix
+			return nil
+		}
+
+		setterNode, err := node.Value.Pipe(yaml.Lookup(setters2.K8sCliExtensionKey, "setter"))
+		if err != nil {
+			return err
+		}
+		if yaml.IsMissingOrNull(setterNode) {
+			// has the setter prefix, but missing the setter extension
+			return errors.Errorf("missing x-k8s-cli.setter for %s", key)
+		}
+
+		// unmarshal the yaml for the setter extension into the definition struct
+		b, err := setterNode.String()
+		if err != nil {
+			return err
+		}
+		if err := yaml.Unmarshal([]byte(b), &setter); err != nil {
+			return err
+		}
+
+		// the description is not part of the extension, and should be pulled out
+		// separately from the extension values.
+		description := node.Value.Field("description")
+		if description != nil {
+			setter.Description = description.Value.YNode().Value
+		}
+		if len(setter.ListValues) > 0 {
+			var vals string
+			for _, val := range setter.ListValues {
+				vals = vals + ", " + val
+			}
+			vals = strings.TrimPrefix(vals, ", ")
+			setters[setter.Name] = fmt.Sprintf("[%s]", vals)
+		} else {
+			setters[setter.Name] = setter.Value
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return setters, nil
+}
+
+// getExtFromComment returns the cliExtension openAPI extension if it is present as
+// a comment on the field.
+func getExtFromComment(schema *openapi.ResourceSchema) (*setters2.CliExtension, error) {
+	if schema == nil {
+		// no schema found
+		// TODO(pwittrock): should this be an error if it doesn't resolve?
```

Comment:
Can we avoid copying code?

Reply:
These methods needed some tweaking when compared to downstream kyaml and also are private there. kpt functions catalog uses v0.10.13 version of kyaml. Upgrading kyaml needs refactor of all functions.